api/src/app/modules/project/public-notice.service.ts

```diff
@@ -104,12 +108,18 @@ export class PublicNoticeService extends DataService<PublicNotice, Repository<Pu
     if (cacheResult != undefined) {
       return cacheResult as PublicNoticePublicFrontEndResponse[];
     }
-
     const query = this.repository.createQueryBuilder("pn")
       .leftJoinAndSelect("pn.project", "p")
       .leftJoinAndSelect("p.forestClient", "forestClient")
       .leftJoinAndSelect("p.district", "district")
-      .andWhere("p.workflow_state_code IN (:...workflowStateCodes)", { workflowStateCodes: [WorkflowStateEnum.COMMENT_OPEN]})
+			// public notices posted earlier than commenting open (in published state with post_date on before today).
+			.where(new Brackets(qb => {
+				qb.where("p.workflow_state_code =:workflowStateCode", {workflowStateCode: WorkflowStateEnum.PUBLISHED})
+				.andWhere("pn.post_date <=:today", {today: DateTimeUtil.nowBC().toDate().toISOString()})
+			}))
+			// public notices for commenting open.
+			.orWhere("p.workflow_state_code IN (:...workflowStateCodes)",
```

Comment:
I'm confused why we check here for COMMENT_OPEN but above we check only for PUBLISHED status. I believe this 'published' is wrong, it should be checking for open. 
(The batch job converts from published to commenting open, only then can they appear in public front-end)

Reply:
Hi Basil. The "PUBLISHED" status is intended. Our ticket this time is to show public-notice at the "public" side even prior to commenting_open (for the Project). And the only time this happens is when public users go to view on public side on (**today**) is between:  **published pushed date ("PUBLISHED")<= post_date (Public can view) <= commenting_open_date ("COMMENTING_OPEN")**.

So while previously we are checking for just COMMENTING_OPEN that's enough, this time, we also want to include this part "published pushed date ("PUBLISHED")<= post_date (Public can view) " for query to find them.

Hope that make sense, it is tested, show and discussed with Julius last week, he confirm that's what he wants to see. If you thinks the query now can be simplify or still wrong, please let me know.

Reply:
Ah, okay that does make sense. It is a shift in how we use the statuses, so we need to remember that going forward. I believe the logic though is if (current date >= post date and status = Published) OR (status = commenting open), then can view the notice. Previously, the batch process ensured that status only goes to commenting open when current date >= commenting open date, so the query doesn't need to check the commenting open date directly. 
So since we don't want to add a new status, we need the current date >= post date when published for the public notices.

So I think it might be okay...

Reply:
:)! Thanks Basil.
Yes, "(current date >= post date and status = Published) **OR** (status = commenting open)", that's why there is a "where(**new Brackets(** first condition... **)** ) **.orWhere**( second condition...)"

Reply:
I re-reviewed and am fine with this code.